### PR TITLE
Head revs cannot deplete flashes.

### PR DESCRIPTION
**Pull request by PowerfulBacon**
## About The Pull Request

closes https://github.com/BeeStation/BeeStation-Hornet/pull/4231
Flashes can still be examined to see how many charges they have left.
Head revs spawn with a normal flash.
When a head rev uses a flash it will gain a charge at a random probability inversely proportional to the number of charges left. (When there is only 1 charge or fewer left, an extra charge will always be added, the less charges there are the higher the chance the flash has to recharge)

## Why It's Good For The Game

Prevents metaing head rev flashes, still allows examining to see how many charges are left on a flash.

## Changelog
:cl:
tweak: Head revs now use a normal flash
tweak: When a head rev uses a flash it will gain charges.
/:cl:

### CODE REVIEW

**Comment by ItsTisme** [comment]
PRetty good fix, me likey. Hard to force some one to flash 10 times to confirm innosence

**Comment by Archanial** [comment]
It doesn't need to close #4231

**Comment by PowerfulBacon** [comment]
It was made specifically to close #4231
I want the uses on flashes to be visible, and that PR only solves the same issue as this PR (Nothing else is mentioned in the WIGFTG) so becomes redundant with this PR.

**Comment by bloons3** [comment]
Shouldn't they spawn with the same flash that could be printed from robotics?
Otherwise there is still a meta element that their flash isn't "normal".

**Comment by OceanFish1** [comment]
Thats much better than just removing numbers on examine

**Comment by francinum** [comment]
This could honestly be a trait granted to headrevs instead of attached directly to the headrev status, but seems fine.

**Comment by qwertyquerty** [comment]
I'm perfectly onboard with the idea of just giving headrevs a completely normal flash. Dunno what other people think of this.

**Comment by bloons3** [comment]
What was wrong with the old system of "flashes only burn out if you use them too quickly"?

**Comment by PowerfulBacon** [comment]
> 
> 
> What was wrong with the old system of "flashes only burn out if you use them too quickly"?

This pr doesn't affect flashes in general only head flashes.

File: code/modules/antagonists/revolution/revolution.dm [hunk 1]
```diff
@@ -269,7 +269,7 @@
 		H.dna.remove_mutation(CLOWNMUT)
 
 	if(give_flash)
-		var/obj/item/assembly/flash/handheld/strong/T = new(H)
+		var/obj/item/assembly/flash/T = new(H)
```
Comment (PowerfulBacon) [inline code/modules/antagonists/revolution/revolution.dm]:
```suggestion
		var/obj/item/assembly/flash/handheld/T = new(H)
```

**Review by utilbird — APPROVED** [review]

**Pull request merged**